index.js

```diff
@@ -0,0 +1,196 @@
+  type: 'string',
+  default: 'at 12:00am',
+  alias: 'interval'
+})
+.option('z', {
+  description: 'current timezone in which interval takes place',
+  default: 'America/Los_Angeles',
+  type: 'string',
+  alias: 'timezone'
+})
+.option('r', {
+  description: 'immediately purge docker instances instead of scheduling',
+  default: false,
+  type: 'boolean',
+  alias: 'runNow'
+})
+.option('s', {
+  description: 'swarm mode, when purging a container will also purge any docker services in its swarm',
+  type: 'boolean',
+  default: false,
+  alias: 'swarm'
+})
+.option('I', {
+  description: 'only remove containers matching this (using JS RegEx notation) ',
+  default: false,
+  alias: 'include'
+})
+.option('e', {
+  description: 'do not remove containers matching this (using JS RegEx notation) ',
+  default: false,
+  alias: 'exclude'
+})
+.help('h', 'help')
+.argv;
+
+const later = require('later');
+modifyLater(later, argv.timezone);
+
+const colors = {
+  start: 'bgGreen',
+  stop: 'bgRed'
+};
+
+const logOptions = {
+  reporters: {
+    flat: {
+      reporter: require('logr-flat'),
+      options: {
+        timestamp: false,
+        appColor: true,
+        colors
+      }
+    }
+  }
+};
+
+const log = Logr.createLogger(logOptions);
+log(`will halt all docker instances older than ${argv.days} days at ${argv.interval}, ${argv.timezone} timezone`);
+const isExpired = (expirationCutoff, containerCreated) => new Date().getTime() - (containerCreated * 1000) > expirationCutoff;
+
+const docker = new Dockerode();
+
+const purge = (argv) => {
+  log(`Purging outstanding docker containers at ${new Date()}`);
+  const expirationLimit = argv.days * 24 * 60 * 60 * 1000;
+
+  async.autoInject({
+    // in swarm mode, all services are part of the swarm and should be killed first:
+    services(done) {
+      if (!argv.swarm) {
+        return done();
+      }
+      docker.listServices(done);
+    },
+    expiredServices(services, done) {
+      if (!argv.swarm) {
+        return done();
+      }
+      const expiredServices = [];
+      services.forEach((serviceInfo) => {
+        const serviceCreated = (new Date().getTime() - new Date(serviceInfo.CreatedAt).getTime()) / 1000;
+        if (isExpired(expirationLimit, serviceCreated)) {
+          expiredServices.push(docker.getService(serviceInfo.ID));
+        }
+      });
+      return done(null, expiredServices);
+    },
+    removeServices(expiredServices, done) {
+      if (!argv.swarm) {
+        return done();
+      }
+      async.eachSeries(expiredServices, (expired, eachDone) => {
+        expired.remove((err) => {
+          if (err) {
+            log(err);
+          }
+          log(['removed'], { id: expired.Name });
+          eachDone();
+        });
+      }, done);
+    },
+
+    // once services are removed, kill relevant containers:
+    list(done) {
+      docker.listContainers(done);
```

Comment:
lets move swarm and containers into separate files and then just conditionally call them based on `--swarm` flag